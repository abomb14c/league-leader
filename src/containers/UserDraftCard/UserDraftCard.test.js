import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import  { UserDraftCard, 
  mapDispatchToProps, 
  mapStateToProps}  
  from './UserDraftCard';

describe('UserDraftCard', () => {
  let wrapper;
  let mockHandleTeam;
  let mockProps;
  
  beforeEach(() => {
    mockHandleTeam = jest.fn();
    mockProps = {
      EPL: [{}, {}, {}],
      user: {user_id: 2}
    };
  
    wrapper = shallow(<UserDraftCard 
      handleTeam={mockHandleTeam}
      {...mockProps}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has a default state", () => {
    let expected = {
      teams: []
    };

    expect(wrapper.state()).toEqual(expected);
  });

  it('handleDrop should set the state of teams', () => {
    
    const teamData = [{}, {}, {}];

    wrapper.instance().handleDrop(teamData);
    wrapper.setState({teams: [{}, {}, {}]});
    expect(wrapper.state()).toEqual({teams: [{}, {}, {}]});
  });

  it('handleDrop should call handleTeam', () => {

    const teamData = [{}, {}, {}];

    wrapper.instance().handleDrop(teamData);
    expect(mockHandleTeam).toHaveBeenCalledWith(teamData);
  });

  it('handleDragover should have the correct drop effect', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
      dataTransfer: { dropEffect: "move" }
    };

    wrapper.instance().handleDragover(mockEvent);

    expect(mockEvent.dataTransfer.dropEffect).toEqual('move');
  });
});


describe('mapStateToProps', () => {
  it('should map the user and EPL to props', () => {
    const mockState = {
      user: {user_id:1},
      EPL: [{}, {}, {}],
      draftTeams: [{}, {}, {}]
    };

    const expected = {
      user: {user_id:1}
    };
    
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});

describe('mapDispatchtoProps', () => {
  it('handleTeam should be called with the correct params', async () => {

    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);

    const team = {};

    const mockAction = {
      type: "REMOVE_TEAM", 
      team
    };


    mappedProps.handleTeam(team);

    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  });
});