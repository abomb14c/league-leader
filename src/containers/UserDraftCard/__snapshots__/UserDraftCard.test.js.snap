// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDraftCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDraftCard
    EPL={
        Array [
            Object {},
            Object {},
            Object {},
          ]
    }
    handleTeam={[Function]}
    user={
        Object {
            "user_id": 2,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="user-title-container"
>
          <h3
                    className="user-draft-title"
          >
                    2
          </h3>
</div>,
        <Droppable
          enabled={true}
          onDrop={[Function]}
          types={
                    Array [
                              "team",
                            ]
          }
>
          <div
                    className="draft-board"
                    id="target"
                    onDragOver={[Function]}
          >
                    <DraftPicks
                              teams={Array []}
                    />
          </div>
</Droppable>,
      ],
      "className": "user-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3
            className="user-draft-title"
>
            2
</h3>,
          "className": "user-title-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "className": "user-draft-title",
          },
          "ref": null,
          "rendered": 2,
          "type": "h3",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="draft-board"
            id="target"
            onDragOver={[Function]}
>
            <DraftPicks
                        teams={Array []}
            />
</div>,
          "enabled": true,
          "onDrop": [Function],
          "types": Array [
            "team",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <DraftPicks
              teams={Array []}
/>,
            "className": "draft-board",
            "id": "target",
            "onDragOver": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "teams": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="user-title-container"
>
            <h3
                        className="user-draft-title"
            >
                        2
            </h3>
</div>,
          <Droppable
            enabled={true}
            onDrop={[Function]}
            types={
                        Array [
                                    "team",
                                  ]
            }
>
            <div
                        className="draft-board"
                        id="target"
                        onDragOver={[Function]}
            >
                        <DraftPicks
                                    teams={Array []}
                        />
            </div>
</Droppable>,
        ],
        "className": "user-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3
              className="user-draft-title"
>
              2
</h3>,
            "className": "user-title-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "className": "user-draft-title",
            },
            "ref": null,
            "rendered": 2,
            "type": "h3",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="draft-board"
              id="target"
              onDragOver={[Function]}
>
              <DraftPicks
                            teams={Array []}
              />
</div>,
            "enabled": true,
            "onDrop": [Function],
            "types": Array [
              "team",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <DraftPicks
                teams={Array []}
/>,
              "className": "draft-board",
              "id": "target",
              "onDragOver": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "teams": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
